core/src/com/unciv/models/ruleset/unique/UniqueTriggerActivation.kt

```diff
@@ -115,12 +119,21 @@ object UniqueTriggerActivation {
 
                 val tilesUnitsWerePlacedOn: MutableList<Vector2> = mutableListOf()
                 repeat(actualAmount) {
-                    val placedUnit = if (city != null || tile == null) civInfo.units.addUnit(unit, chosenCity)
-                        else civInfo.units.placeUnitNearTile(tile.position, unit)
+                    val placedUnit = when {
```

Comment:
Hm duplocated code, maybe extract to small function? Not a sticking point just a thought